### PR TITLE
include/rdma: improve API

### PR DESCRIPTION
**Pull request by aingerson**

### CODE REVIEW

**Comment by shefty** [comment]
This. Is. Beautiful.  Now, where's that dancing monkey?

File: include/rdma/fi_atomic.h [hunk 1]
```diff
@@ -53,83 +53,83 @@ struct fi_atomic_attr {
...
 	void			*context;
 	uint64_t		data;
 };
 
 struct fi_msg_fetch {
-	struct fi_ioc		*msg_iov;
+	struct fi_ioc		*msgIov;
 	void			**desc;
-	size_t			iov_count;
+	size_t			iovcount;
```
Comment (shefty) [inline include/rdma/fi_atomic.h:71]: C should be upper case.

File: include/rdma/fi_atomic.h [hunk 2]
```diff
@@ -147,147 +147,147 @@ struct fi_ops_atomic {
...
 	return ep->atomic->compwritemsg(ep, msg,
-			comparev, compare_desc, compare_count,
-			resultv, result_desc, result_count, flags);
+			comparev, compareDesc, compareCount,
+			resultv, resultDesc, resultCount, flags);
 }
 
 static inline int
-fi_atomicvalid(struct fid_ep *ep,
+fiAtomicvalid(struct fid_ep *ep,
```
Comment (shefty) [inline include/rdma/fi_atomic.h]: V should be upper case.  Comment applies to other calls.

File: include/rdma/fi_atomic.h [hunk 3]
```diff
@@ -147,147 +147,147 @@ struct fi_ops_atomic {
...
-			resultv, result_desc, result_count,
-			dest_addr, addr, key, datatype, op, context);
+			comparev, compareDesc, compareCount,
+			resultv, resultDesc, resultCount,
+			destAddr, addr, key, datatype, op, context);
 }
 
 static inline ssize_t
-fi_compare_atomicmsg(struct fid_ep *ep,
+fiCompareAtomicmsg(struct fid_ep *ep,
```
Comment (shefty) [inline include/rdma/fi_atomic.h]: M in Msg should be upper case.  Comment applies elsewhere.

**Review by shefty — CHANGES_REQUESTED** [review]

**Comment by j-xiong** [comment]
Good one!

**Pull request closed without merging**